### PR TITLE
feat: Add pub/sub ordering keys examples

### PR DESCRIPTION
**Pull request by kamalaboulhosn**
Adds the following region tags:

- `pubsub_enable_subscription_ordering`
- `pubsub_publish_with_ordering_keys`
- `pubsub_resume_publish_with_ordering_keys`

closes: #651

### CODE REVIEW

File: pubsub/topics.rb
Comment (quartzmo) [inline pubsub/topics.rb:443]: An extra space in this line needs to be removed to fix column alignment.

**Review by quartzmo — COMMENTED** [review]
Additions to `topics.rb` should be accompanied by matching tests in [spec/topics_spec.rb](https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/master/pubsub/spec/topics_spec.rb).

File: pubsub/topics.rb
Comment (kamalaboulhosn) [reply pubsub/topics.rb]: Done.

File: pubsub/topics.rb
Comment (kamalaboulhosn) [reply pubsub/topics.rb]: Done.

**Comment by kamalaboulhosn** [comment]
@quartzmo I have added tests to the Pub/Sub samples. Thanks!

**Comment by quartzmo** [comment]
CI failure is in firestore and is unrelated. The specs for pubsub passed.

File: pubsub/topics.rb
Comment (quartzmo) [inline pubsub/topics.rb:360]: [AsyncPublisher#stop!](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-pubsub/lib/google/cloud/pubsub/async_publisher.rb#L183-L197) combines these two methods, if you would prefer to show it instead. In google-cloud-ruby's inline api samples, `stop!` is shown more than `stop` and `wait!`, but there are still 15 usages of the latter, so I suppose either is OK for now.

File: pubsub/topics.rb
Comment (quartzmo) [inline pubsub/topics.rb:327]: [AsyncPublisher#stop!](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-pubsub/lib/google/cloud/pubsub/async_publisher.rb#L183-L197) combines these two methods, if you would prefer to show it instead. In google-cloud-ruby's inline api samples, `stop!` is shown more than `stop` and `wait!`, but there are still 15 usages of the latter, so I suppose either is OK for now.

File: pubsub/topics.rb
Comment (quartzmo) [inline pubsub/topics.rb:354]: Usually operations are pretty self-explanatory, but I wonder if `resume_publish` could use a comment here explaining what it does?

**Review by quartzmo — APPROVED** [review]
Looks good! I posted a couple of questions, but feel free to merge.

File: pubsub/topics.rb
Comment (kamalaboulhosn) [reply pubsub/topics.rb]: Done.

File: pubsub/topics.rb
Comment (kamalaboulhosn) [reply pubsub/topics.rb:356]: Done.

**Comment by kamalaboulhosn** [comment]
Fixed the tags as @anguillanneuf suggested. @quartzmo I don't have write access to this repo so can you merge it? Thanks!

**Comment by quartzmo** [comment]
I am also unable to merge PRs in this repo unless the CI build passes. I believe it is failing due to an unrelated issue:

```
+ bundle exec rubocop
Inspecting 476 files
.....................................................................................................................................................................................................................................................................................................................................................................................................................C..............C.......................................................

Offenses:

functions/slack/app.rb:45:1: C: Layout/CommentIndentation: Incorrect indentation detected (column 0 instead of 2).
# [START functions_verify_webhook]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
functions/slack/app.rb:56:18: C: Style/PerlBackrefs: Avoid the use of Perl-style backrefs.
          result[$1.split("_").map(&:capitalize).join("-")] = val
                 ^^
functions/slack/app.rb:75:1: C: Layout/CommentIndentation: Incorrect indentation detected (column 0 instead of 2).
# [START functions_slack_request]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
functions/slack/app.rb:85:1: C: Layout/CommentIndentation: Incorrect indentation detected (column 0 instead of 2).
# [START functions_slack_format]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
functions/slack/app.rb:107:1: C: Layout/CommentIndentation: Incorrect indentation detected (column 0 instead of 2).
# [END functions_slack_format]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
functions/test/slack_test.rb:27:19: C: Style/BlockDelimiters: Prefer do...end for multi-line blocks without chaining.
  let(:signature) {
                  ^
functions/test/slack_test.rb:37:21: C: Style/BlockDelimiters: Prefer do...end for multi-line blocks without chaining.
  let(:kg_response) {
                    ^
functions/test/slack_test.rb:55:24: C: Style/BlockDelimiters: Prefer do...end for multi-line blocks without chaining.
  let(:slack_response) {
                       ^
functions/test/slack_test.rb:69:30: C: Style/BlockDelimiters: Prefer do...end for multi-line blocks without chaining.
  let(:empty_slack_response) {
                             ^
functions/test/slack_test.rb:102:25: C: Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments.
      mock_client.expect(:search_entities, kg_response, [query: query, limit: 1])
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

476 files inspected, 10 offenses detected
```

@dazuma Are you able to merge this PR? If so, can you please?

**Pull request merged**
